web/packages/teleport/src/Assist/Chat/ChatItem/Action/RunAction.tsx

```diff
@@ -0,0 +1,209 @@
+import { MessageTypeEnum, Protobuf } from 'teleport/lib/term/protobuf';
+import { Dots } from 'teleport/Assist/Dots';
+
+interface RunCommandProps {
+  actions: ExecuteRemoteCommandContent;
+}
+
+function convertContentToCommand(message: ExecuteRemoteCommandContent) {
+  const command = {
+    command: '',
+    login: '',
+    query: '',
+  };
+
+  if (message.selectedLogin) {
+    command.login = message.selectedLogin;
+  }
+
+  if (message.command) {
+    command.command = message.command;
+  }
+
+  if (message.query) {
+    command.query = message.query;
+  }
+
+  return command;
+}
+
+enum RunActionStatus {
+  Connecting,
+  Finished,
+}
+
+interface NodeState {
+  nodeId: string;
+  status: RunActionStatus;
+  stdout?: string;
+}
+
+interface RawPayload {
+  node_id: string;
+  payload: string;
+}
+
+export function RunCommand(props: RunCommandProps) {
+  const { clusterId } = useStickyClusterId();
+  const urlParams = useParams<{ conversationId: string }>();
+
+  const [state, setState] = useState(() => []);
+
+  const params = convertContentToCommand(props.actions);
+
+  const execParams = {
+    ...params,
+    conversation_id: urlParams.conversationId,
+    execution_id: crypto.randomUUID(),
+  };
+
+  const search = new URLSearchParams();
+
+  search.set('access_token', getAccessToken());
+  search.set('params', JSON.stringify(execParams));
+
+  const url = `wss://${getHostName()}/v1/webapi/command/${clusterId}/execute?${search.toString()}`;
+
+  const websocket = useRef<WebSocket>(null);
+  const protoRef = useRef<any>(null);
+
+  useEffect(() => {
+    if (!websocket.current) {
+      const proto = new Protobuf();
+      const ws = new WebSocket(url);
+      ws.binaryType = 'arraybuffer';
+
+      ws.onmessage = event => {
+        const uintArray = new Uint8Array(event.data);
+        const msg = proto.decode(uintArray);
+
+        switch (msg.type) {
+          case MessageTypeEnum.SESSION_DATA:
+            break;
+
+          case MessageTypeEnum.RAW:
+            const data = JSON.parse(msg.payload) as RawPayload;
+            const payload = atob(data.payload);
+
+            setState(state => {
+              const results = state.find(node => node.nodeId == data.node_id);
+              if (!results) {
+                state.push({
+                  nodeId: data.node_id,
+                  status: RunActionStatus.Connecting,
+                });
+              }
+
+              const s = state.map(item => {
+                if (item.nodeId === data.node_id) {
+                  if (!item.stdout) {
+                    item.stdout = '';
+                  }
+                  return {
+                    ...item,
+                    status: RunActionStatus.Finished,
+                    stdout: item.stdout + payload,
+                  };
+                }
+
+                return item;
+              });
+
+              return s;
+            });
+
+            break;
+
+          default:
+            break;
```

Comment:
probably want to log this for later like we do here: https://github.com/gravitational/teleport/blob/89fdad50d011cdbcc8e719fd3bdfdc3fd9e99a7c/web/packages/teleport/src/lib/term/tty.ts#L176

Reply:
I've just removed the default/empty case statement for now, I don't think this needs logging (as this `default` is handling all the other `MessageTypeEnum` values)